src/software/ai/hl/stp/tactic/attacker/attacker_tactic.h

```diff
@@ -17,18 +36,48 @@ class AttackerTactic : public Tactic
     std::string getFSMState() const override;
 
     /**
-     * Evaluate the AttackerTactic and start a new iteration
+     * Terminate the current episode of the AttackerTactic and reset the tactic
+     * for a new episode
      *
-     * @param score score between [-1, 1] rating the success of the AttackerTactic
-     * for the current iteration
+     * @param world_ptr the World at the end of the current episode
      */
-    void evaluate(double score);
+    void terminate(const WorldPtr& world);
 
    private:
-    std::shared_ptr<Strategy> strategy;
+    // The shared Strategy used by all of AI
+    std::shared_ptr<Strategy> strategy_;
+
+    // The Q-function of the attacker agent's policy
+    std::shared_ptr<LinearQFunction<AttackerMdpState, AttackerMdpAction>> q_function_;
+
+    // The action selection strategy of attacker agent's policy
+    std::shared_ptr<EpsilonGreedyStrategy<AttackerMdpState, AttackerMdpAction>>
+        action_selection_strategy_;
+
+    // The policy that the attacker agent will follow
+    QPolicy<AttackerMdpState, AttackerMdpAction> policy_;
+
+    // The reward function for the attacker agent
+    GameplayMonitor gameplay_monitor_;
 
-    SkillGraph skill_graph_;
-    std::shared_ptr<Skill> current_skill_;
+    // The current skill that the attacker is executing
+    std::unique_ptr<Skill> current_skill_;
+
+    // File to load Q-function weights from for initialization
+    inline static const std::string ATTACKER_MDP_Q_FUNCTION_INITIAL_WEIGHTS_FILE =
+        "software/ai/evaluation/q_learning/attacker_mdp_q_function_weights.csv";
+
+    // Name of file that Q-function weights will be saved to at runtime
+    inline static const std::string ATTACKER_MDP_Q_FUNCTION_RUNTIME_WEIGHTS_FILE =
+        "attacker_mdp_q_function_weights.csv";
```

Comment:
could be a constexpr if you use a `char*` array instead

Reply:
looks like using char array or `std::string_view` will construct a new `std::string` each time https://stackoverflow.com/questions/27123306/is-it-possible-to-use-stdstring-in-a-constant-expression